### ISSUE TITLE
Redirect UI: Separate internal paths/aliases vs external URLs.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1803]
This is a follow-up to #905 (as per https://github.com/backdrop/backdrop/pull/1317#issuecomment-210282347)...

![screen shot 2016-04-14 at 8 44 12 pm](https://cloud.githubusercontent.com/assets/397895/14550880/bf9b4346-0281-11e6-83fa-78714b1d4e0a.png)

**Comment by quicksketch** [comment]
So the complaint here is that the "From" field is prefixed with the site URL, but the "To" field is not. This is because the "To" field can also be a full URL. e.g. You could redirect from

From:
`http://example.com/` [`microsite`]
To:
[`http://microsite.example.com`]

Options for fixing this include:

- Removing the site URL prefix from the "From" field to make it match "To".
- Adding the site URL prefix to the "To" field even though it is inaccurate.
- Remove the ability to have an external redirect entirely (thus making it so we could put a prefix on "To" accurately).
- Show a prefix conditionally. Based on either asking the user whether it is external or by determining it automatically.

I don't think any of these are very good options.

**Comment by BWPanda** [comment]
I agree that none of the suggested options are very good. Another not-so-good option would be to have two 'To' fields: one for internal URLs (prefixed) and one for external (not prefixed). One would have to take preference over the other in case both are filled.

Maybe instead all that's needed is better documentation on this screen...?

**Comment by BWPanda** [comment]
FTR, the Link Destination field in CKEditor's Link popup is for both local and external URLs, and it's un-prefixed, and it's an autocomplete :wink: 

![link](https://user-images.githubusercontent.com/2385329/58584260-95023f80-825e-11e9-8d15-f7131dd96d8f.jpg)

**Comment by klonos** [comment]
Yeah, I see that conditionally adding the prefix in the "To" field seems like it cannot be solved in any elegant way.

I like that idea of having the from/to fields be autocompletes, but most likely close this issue here as a `wontfix`, and file a separate one for this.

We should also make the front page selection field an autocomplete btw (#1628)
